lib/providers/gcm.js

```diff
@@ -79,13 +79,16 @@ GcmProvider.prototype._createMessage = function(notification) {
     delayWhileIdle: notification.delayWhileIdle,
   });
 
-  var propNames = Object.keys(notification);
-  // GCM does not have reserved message parameters for alert or badge, adding them as data.
-  propNames.push('alert', 'badge');
+  // Dont send these keys in the message payload
+  var exclusions = ['id', 'deviceToken', 'deviceType']
```

Comment:
Since support for pre-v4 versions of Node.js has been already dropped, we should use `const` instead of `var`. See http://loopback.io/doc/en/contrib/style-guide.html#variable-declarations